R/features.R

```diff
@@ -16,22 +16,33 @@
 #' @keywords internal
 #'
 #' @author Nikolai Sellereite, Martin Jullum
-feature_combinations <- function(m, exact = TRUE, noSamp = 200, shapley_weight_inf_replacement = 10^6, reduce_dim = TRUE) {
+feature_combinations <- function(m, exact = TRUE, n_combinations = 200, shapley_weight_inf_replacement = 10^6, reduce_dim = TRUE) {
+
+  # Force user to use a natural number for n_combinations if m > 10
+  if (m > 10 & is.null(n_combinations)) {
+    stop(
+      paste(
+        "If the number of features is greater than 10, you need to choose a natural number\n ",
+        "for argument n_combinations. You can still use the exact method by setting\n ",
+        "n_combinations equal to 2^m, where m is the number of features."
+      )
+    )
+  }
 
   # Not supported for m > 30
   if (m > 30) {
-    stop("Currently we are not supporting cases where m > 30.")
+    stop("Currently we are not supporting cases where the number of features is greater than 30.")
   }
 
-  if (!exact && noSamp > (2^m - 2) && !reduce_dim) {
-    noSamp <- 2^m - 2
-    cat(sprintf("noSamp is larger than 2^m = %d. Using exact instead.", 2^m))
+  if (!exact && n_combinations > (2^m - 2) && !reduce_dim) {
+    n_combinations <- 2^m - 2
+    cat(sprintf("n_combinations is larger than 2^m = %d. Using exact instead.", 2^m))
```

Comment:
It might be confusing when you are told to set n_combinations = 2^m, then this warning tells you that you have used MORE than 2^m.
Suggests:
cat(sprintf("n_combinations is larger than or equal to 2^m = %d. Using exact instead.", 2^m))

Reply:
Fixed with fa8ec7b.